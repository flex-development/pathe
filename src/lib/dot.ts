/**
 * @file dot
 * @module pathe/lib/dot
 */

import type { Dot } from '@flex-development/pathe'

/**
 * Dot character (`'.'`).
 *
 * @see {@linkcode Dot}
 *
 * @category
 *  utils
 *
 * @const {Dot} dot
 */
const dot: Dot = '.'

export default dot
