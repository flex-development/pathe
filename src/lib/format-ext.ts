/**
 * @file formatExt
 * @module pathe/lib/formatExt
 */

import validateString from '#internal/validate-string'
import type { EmptyString, Ext } from '@flex-development/pathe'

/**
 * Format a file extension.
 *
 * @see {@linkcode EmptyString}
 * @see {@linkcode Ext}
 *
 * @example
 *  formatExt('') // ''
 * @example
 *  formatExt(null) // ''
 * @example
 *  formatExt('.ts ') // '.ts'
 * @example
 *  formatExt(' mjs') // '.mjs'
 * @example
 *  formatExt('d.mts') // '.d.mts'
 *
 * @category
 *  utils
 *
 * @param {string | null | undefined} ext
 *  File extension to format
 * @return {EmptyString | Ext}
 *  Formatted file extension or empty string
 */
function formatExt(ext: string | null | undefined): EmptyString | Ext {
  if (ext !== null && ext !== undefined) {
    validateString(ext, 'ext')
    ext = ext.trim()
  }

  if (!ext) return ''
  if (ext.startsWith('.')) return <Ext>ext
  return `.${ext}`
}

export default formatExt
