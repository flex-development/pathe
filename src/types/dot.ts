/**
 * @file Type Aliases - Dot
 * @module pathe/types/Dot
 */

/**
 * A dot character (`'.'`).
 */
type Dot = '.'

export type { Dot as default }
