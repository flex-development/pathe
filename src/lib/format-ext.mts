/**
 * @file formatExt
 * @module pathe/lib/formatExt
 */

import validateString from '#internal/validate-string'
import dot from '#lib/dot'
import type { EmptyString, Ext } from '@flex-development/pathe'

/**
 * Format a file extension.
 *
 * @see {@linkcode EmptyString}
 * @see {@linkcode Ext}
 *
 * @example
 *  formatExt('') // ''
 * @example
 *  formatExt(null) // ''
 * @example
 *  formatExt('.ts ') // '.ts'
 * @example
 *  formatExt(' mjs') // '.mjs'
 * @example
 *  formatExt('d.mts') // '.d.mts'
 *
 * @category
 *  utils
 *
 * @this {void}
 *
 * @param {string | null | undefined} ext
 *  The file extension to format
 * @return {EmptyString | Ext}
 *  Formatted file extension or empty string
 */
function formatExt(
  this: void,
  ext: string | null | undefined
): EmptyString | Ext {
  if (ext !== null && ext !== undefined) {
    validateString(ext, 'ext')
    ext = ext.trim()
  }

  if (!ext) return ''
  if (ext.startsWith(dot)) return ext as Ext
  return `${dot}${ext}`
}

export default formatExt
