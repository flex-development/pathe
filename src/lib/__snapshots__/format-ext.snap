// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:lib/formatExt > should return formatted file extension (" ") 1`] = `""`;

exports[`unit:lib/formatExt > should return formatted file extension (" mjs") 1`] = `".mjs"`;

exports[`unit:lib/formatExt > should return formatted file extension ("") 1`] = `""`;

exports[`unit:lib/formatExt > should return formatted file extension (".ts ") 1`] = `".ts"`;

exports[`unit:lib/formatExt > should return formatted file extension ("d.mts") 1`] = `".d.mts"`;

exports[`unit:lib/formatExt > should return formatted file extension (null) 1`] = `""`;
